databricks/koalas/tests/test_indexes.py

```diff
@@ -278,6 +278,17 @@ def test_dropna(self):
         self.assert_eq(kidx.dropna(), pidx.dropna())
         self.assert_eq((kidx + 1).dropna(), (pidx + 1).dropna())
 
+    def test_putmask(self):
+        pidx = pd.Index(["a", "b", "c", "d", "e"])
+        kidx = ks.from_pandas(pidx)
+
+        mask = [True if x < 2 else False for x in range(5)]
```

Comment:
Let's add more test cases for `mask` and `value`.

`value` also can be an Index or a Series, and even can be a list or a tuple. 🤪

Reply:
I will modify it to work in the cases mentioned above as well :)

Reply:
@itholic I modified the `mask` and `value` parameters to support `Series` and `Index`
and also `list` and `tuple`.
However, in numeric values, there is a problem that the output of the result through the internal spark operation is converted to a string. I am stuck here, 😭 could you help me how to solve this?

```
>>> pidx = pd.Index([1, 2, 3, 4, 5])
>>> kidx = ks.from_pandas(pidx)

>>> kidx.putmask(kidx < 3, [100, 200, 300, 400, 500]).sort_values()
Index(['100', '200', '3', '4', '5'], dtype='object')                            

>>> pidx.putmask(pidx < 3, [100, 200, 300, 400, 500]).sort_values()
Int64Index([3, 4, 5, 100, 200], dtype='int64')
```

Reply:
To prevent data type changes, I modified code to use pyspark's `_infer_type` to infer the data type from `value` parameter. Please check if I can use this function.

Reply:
@itholic I solved the type infernece problem using `as_spark_type(pandas_value.dtype.type)`